tests/sqlparser_redshift.rs

```diff
@@ -353,3 +378,20 @@ fn test_parse_json_path_from() {
         _ => panic!(),
     }
 }
+
+#[test]
+fn test_parse_select_numbered_columns() {
```

Comment:
thinking a couple cases we can include?
```sql
SELECT 1 AS ["1];
SELECT 1 AS [ " 1 " ];
```

Reply:
I split the test into two to make it more explicit:
1. `test_parse_select_numbered_columns` - it is a reason for this MR
2. `test_parse_nested_quoted_identifier`

I added suggested cases into `test_parse_nested_quoted_identifier`